src/core/attributeformmodelbase.cpp

```diff
@@ -546,10 +546,13 @@ void AttributeFormModelBase::updateVisibilityAndConstraints( int fieldIndex )
   {
     if ( fieldIndex == -1 || it.first.referencedAttributeIndexes( fields ).contains( fieldIndex ) )
     {
+      qDebug() << fieldIndex;
       QgsExpression exp = it.first;
+      qDebug() << exp.expression();
       exp.prepare( &mExpressionContext );
 
       bool visible = exp.evaluate( &mExpressionContext ).toInt();
+      qDebug() << ( visible ? "visible" : "not visible" );
```

Comment:
Don't forget your debug messages before merge.

Reply:
@suricactus , whoops, that had nothing to do with this, thanks pointing out my mess ;)